index.html

```diff
@@ -184,96 +174,147 @@
-      if (!event.target.matches('#balance')) return;
-      event.preventDefault();
-      onClick();
     }, false);
 
+    function download(data, filename, type) {
+        var file = new Blob([data], {type: type});
+        if (window.navigator.msSaveOrOpenBlob) // IE10+
+            window.navigator.msSaveOrOpenBlob(file, filename);
+        else { // Others
+            var a = document.createElement("a"),
+                    url = URL.createObjectURL(file);
+            a.href = url;
+            a.download = filename;
+            document.body.appendChild(a);
+            a.click();
+            setTimeout(function() {
+                document.body.removeChild(a);
+                window.URL.revokeObjectURL(url);
+            }, 0);
+        }
+      }
+
+      document.addEventListener('click', async function (event) {
+          if (!event.target.matches('#gen-button')) return;
+          event.preventDefault();
+          var kp = Pact.crypto.genKeyPair();
+          var id = kp.publicKey.substring(0, 6);
+          var priv = `public: ${kp.publicKey}\nsecret: ${kp.secretKey}`
+          var privName = `private-keypair-${id}.kda`
+          var pub = `public: ${kp.publicKey}`
+          var pubName = `public-keypair-${id}.kda`
+          download(priv, privName, "text/plain;charset=utf-8");
+          download(pub, pubName, "text/plain;charset=utf-8");
+      }, false);
+
+      function simpleToggle() {
+        var menuD = document.getElementById("menuDiv");
+        if (menuD.style.display === "none") {
+          menuD.style.display = "block";
+        } else {
+          menuD.style.display = "none";
+        }
+      }
   </script>
 </head>
 <body>
     <div id="main">
-      <div class="container">
-          <h1>Kadena Balance Checker</h1>
-          <div id="subhead">
-            <span class="headlink">Balance Checker</span>
+
+      <img class="graph" src="./img/graph.png" alt="graph">
+      <div class="nav">
+        <h1>Kadena Balance Checker</h1>
+        <div class="container">
+          <!-- <div id="subhead">
+            <span class="headlink current">Balance Checker</span>
             <a class="headlink" href="modules.html">Code Viewer</a>
+          </div> -->
+        </div>
+        <div class="container-right">
+          <div id="subhead" class="subheadButton">
+            <span class="headlink" onclick="simpleToggle()">
+              <div>Menu</div>
+              <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" ><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line></svg>
+            </span>
           </div>
+          <span id="menuDiv" class="headlinkContainer" style="display: none;">
+            <a class="headlinkSite" href="https://transfer.chainweb.com/transfer-create.html">
```

Comment:
What is the reasoning behind absolute links here?  One design goal that we had in mind with these tools was to make them usable from a local clone of the repository so that people don't have to trust the external entity that is hosting the site.  It would be nice to preserve that property.

Reply:
I'd also like that, but since both sites weren't in the same repo, I simply used a link.
Your idea of combining them into a single tools site would solve this and make it easier to add more tools etc

Reply:
Great.  How about we sketch it out in this repository?  https://github.com/kadena-community/kadena-tools

Reply:
done!